modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/actions/ItemActionForm.tsx

```diff
@@ -44,32 +74,38 @@ const MESSAGE_TYPES = [
 	},
 ];
 
-const TYPES = [
+const MODAL_VARIANTS = [
 	{
-		label: Liferay.Language.get('link'),
-		value: 'link',
+		label: Liferay.Language.get('full-screen'),
+		value: 'full-screen',
+	},
+	{
+		label: Liferay.Language.get('lg'),
+		value: 'lg',
+	},
+	{
+		label: Liferay.Language.get('sm'),
+		value: 'sm',
 	},
 ];
 
 const translationExists = ({translations}: {translations: any}) => {
 	return Boolean(Object.keys(translations).find((key) => translations[key]));
 };
 interface IFDSItemActionFormProps {
+	activeTab: number;
 	editing?: boolean;
 	fdsView: FDSViewType;
 	initialValues?: IFDSAction;
 	loadFDSActions: () => void;
 	namespace: string;
-	sections: {
-		ACTIONS: string;
-		EDIT_ITEM_ACTION: string;
-		NEW_ITEM_ACTION: string;
-	};
+	sections: typeof SECTIONS;
 	setActiveSection: (arg: string) => void;
 	spritemap: string;
 }
 
 const ItemActionForm = ({
+	activeTab,
```

Comment:
This is `ItemActionForm.tsx`, the idea was for it to contain form only for item actions. With this structure, we should add new `CreationActionForm.tsx`. `activeTab` is a state in parent component that shifts between those two form.

Is there a reason to refactor the whole structure? If it's code duplication, we might consider extracting common components.

Reply:
There are little differences among types, but in general they share lot of controls. See the [figma designs ](https://www.figma.com/file/ThsIousIfUrvJXgy4bfX3j/LPS-193209-design-how-to-define-generic-actions?type=design&node-id=205-66547&mode=design&t=lSok6SosJ9wQFJdL-0). 
In most cases it is just a matter of toggling the visibility of the components.

I could create separate `<type>ActionForm.tsx`, extract common components and compose from them.

Reply:
As the form is so similar, maybe it does makes sense to have both in the same place. In that case, we would want to rename `ItemActionForm` to simply `ActionForm`.

Reply:
I think keeping a single component is a more sensible option at this moment